The error occurs on [line 376 of the Makefile](https://github.com/samtools/htslib/blob/1.20/Makefile#L376), when it is compiling the DLL:

```Make
hts-$(LIBHTS_SOVERSION).dll hts.dll.a: $(LIBHTS_OBJS)
	$(CC) -shared -Wl,--out-implib=hts.dll.a -Wl,--enable-auto-import -Wl,--exclude-all-symbols $(LDFLAGS) -o $@ -Wl,--whole-archive $(LIBHTS_OBJS) -Wl,--no-whole-archive $(LIBS) -lpthread
```

The errors appear to be related to linking to libcurl and libpsl

```
$ gcc -shared -Wl,--out-implib=hts.dll.a -Wl,--enable-auto-import -Wl,--exclude-all-symbols -fvisibility=hidden  -o hts-3.dll -Wl,--whole-archive kfunc.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o header.o hfile.o hts.o hts_expr.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o region.o sam.o sam_mods.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_stats.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/string_alloc.o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/pack.o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static32x16pr_avx2.o htscodecs/htscodecs/rANS_static32x16pr_avx512.o htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_neon.o htscodecs/htscodecs/rANS_static32x16pr.o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rle.o htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/utils.o  hfile_libcurl.o hfile_gcs.o hfile_s3.o hfile_s3_write.o -Wl,--no-whole-archive /mingw64/lib/libregex.a /mingw64/lib/libsystre.a /mingw64/lib/libtre.a /mingw64/lib/libintl.a /mingw64/lib/libiconv.a /mingw64/lib/libdeflate.a /mingw64/lib/liblzma.a /mingw64/lib/libbz2.a /mingw64/lib/libcrypt32.a /mingw64/lib/libwinpthread.a /mingw64/lib/libcurl.a /mingw64/lib/libcrypto.a /mingw64/lib/libssl.a /mingw64/lib/libwldap32.a /mingw64/lib/libcrypto.a /mingw64/lib/libcrypt32.a /mingw64/lib/libcertcli.a /mingw64/lib/libzstd.a /mingw64/lib/libbrotlidec.a /mingw64/lib/libbrotlienc.a /mingw64/lib/libbrotlicommon.a /mingw64/lib/libssh2.a /mingw64/lib/libidn2.a /mingw64/lib/libpsl.a -lws2_32 /mingw64/lib/libunistring.a /mingw64/lib/libnghttp2.a /mingw64/lib/libz.a /mingw64/lib/libbcrypt.a  -lpthread
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xb7a7): undefined reference to `nghttp3_conn_resume_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x1a1a8): undefined reference to `nghttp3_conn_del'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x2de4c): undefined reference to `nghttp3_conn_shutdown_stream_read'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x2de61): undefined reference to `nghttp3_conn_close_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x2de73): undefined reference to `nghttp3_conn_set_stream_user_data'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x312da): undefined reference to `nghttp3_rcbuf_get_buf'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x312f9): undefined reference to `nghttp3_rcbuf_get_buf'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x60c1a): undefined reference to `nghttp3_conn_read_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x60ee3): undefined reference to `nghttp3_strerror'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x60f88): undefined reference to `nghttp3_conn_close_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x60fa4): undefined reference to `nghttp3_strerror'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x60fd1): undefined reference to `nghttp3_conn_close_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x61b61): undefined reference to `nghttp3_conn_writev_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x6201c): undefined reference to `nghttp3_conn_block_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x620f2): undefined reference to `nghttp3_conn_add_write_offset'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x62116): undefined reference to `nghttp3_conn_add_ack_offset'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x62453): undefined reference to `nghttp3_strerror'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x62492): undefined reference to `nghttp3_strerror'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x624ad): undefined reference to `nghttp3_strerror'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x6268e): undefined reference to `nghttp3_conn_unblock_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x62a52): undefined reference to `nghttp3_conn_resume_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x86638): undefined reference to `nghttp3_conn_resume_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x868fb): undefined reference to `nghttp3_conn_submit_request'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x86eb7): undefined reference to `nghttp3_strerror'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x8aa39): undefined reference to `nghttp3_version'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0x8ac54): undefined reference to `nghttp3_version'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xb56cd): undefined reference to `nghttp3_settings_default_versioned'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xb56d2): undefined reference to `nghttp3_mem_default'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xb5700): undefined reference to `nghttp3_conn_client_new_versioned'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xb58fe): undefined reference to `nghttp3_conn_bind_control_stream'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xb5925): undefined reference to `nghttp3_conn_bind_qpack_streams'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xbecbf): undefined reference to `nghttp3_version'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libcurl.a(unity_0_c.c.obj):(.text+0xbecef): undefined reference to `nghttp3_version'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libpsl.a(psl.c.obj):(.text+0x55e): undefined reference to `idn2_lookup_u8'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/lib/libpsl.a(psl.c.obj):(.text+0xab9): undefined reference to `idn2_lookup_u8'
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:376: hts-3.dll] Error 1
##[error]Process completed with exit code 2.
```

--- a/.github/workflows/msys2-htslib-build.yml
+++ b/.github/workflows/msys2-htslib-build.yml
@@ -12,7 +12,7 @@ on:
       - 'scripts/**'
   workflow_dispatch:
 env:
-  PACKAGE_VERSION: 1.19.1
+  PACKAGE_VERSION: "1.20"
   LIBHTS_SOVERSION: 3
   RELEASE_VERSION: 0 # equivalent to conda build number
 jobs:

--- a/README.md
+++ b/README.md
@@ -2,12 +2,13 @@
 
 The workflow [msys2-htslib-build.yml][] builds a statically linked[^explanation] htslib.dll
 under msys2. See the [releases][] to download the tarball. To build a different
-version of htslib, update the env var `PACKAGE_VERSION`. To build an updated
+version of [htslib][], update the env var `PACKAGE_VERSION`. To build an updated
 release of an existing version, update the env var `RELEASE_VERSION`. Also, make
 sure that `RELEASE_VERSION` is reset to 0 when bumping the version.
 
 [msys2-htslib-build.yml]: https://github.com/TileDB-Inc/m2w64-htslib-build/blob/main/.github/workflows/msys2-htslib-build.yml
 [releases]: https://github.com/TileDB-Inc/m2w64-htslib-build/releases
+[htslib]: https://github.com/samtools/htslib
 
 The release tarball is subsequently consumed by
 [TileDB-Inc/m2w64-htslib-feedstock][], which repackages the pre-built files into

